### PR TITLE
Clarify CJALR operation order

### LINKED ISSUE #9: Order of checks for CJALR & invalid address handling not specified
The spec for CJALR says

> The target capability may have Invalid address conversion performed and is then installed in pcc.

but it doesn't say whether invalid address conversion occurs before or after bounds checking (or the other checks, but those are unlikely to be affected by invalid address conversion).

IMO it would be more elegant to do it afterwards - it means most of the CJALR implementation doesn't need to care about invalid addresses; it's all neatly contained in the bit at the end where you install the cap in PCC. Also I imagine it would simplify implementations because there's no data dependency - you can do the bounds check without having to wait for invalid address conversion.

The security properties are the same, since invalid address conversion clears the tag if necessary. The only observable difference would be that in some cases you would get a tag violation on the next cycle, instead of a length violation on the current cycle. I think that's ok though.

### PR DESCRIPTION
**Pull request by Timmmm**
Clarify that length violations are checked on the address after it has undergone invalid address conversion.

Also fix a typo in the title.

Fixes #9

### CODE REVIEW

File: src/insns/cjalr_jalr_32bit.adoc [hunk 1]
```diff
@@ -24,11 +24,20 @@ Capability Mode Description::
...
-of the next instruction following the jump (<<pcc>> + 4) is sealed and written
-to `cd`.
+performed. The check for length violation is done next using the converted
+address and the original bounds.
+
+[NOTE]
+====
+The invalid address conversion representability check can be skipped because the
+length violation check is a stricter condition. The sealing check can also be
+skipped because the capability will always have been unsealed.
```
Comment (arichardson) [inline src/insns/cjalr_jalr_32bit.adoc:36]: I'm not sure about the sealing part? We still want to raise an error if `cs1` is a sentry and the offset is non-zero?

Comment (andresag01) [reply src/insns/cjalr_jalr_32bit.adoc:36]: Agreed, we would like to keep the sealing check

Comment (Timmmm) [reply src/insns/cjalr_jalr_32bit.adoc:36]: I meant the sealing check in the invalid address conversion procedure. I'll reword it to make that clearer.

Comment (Timmmm) [reply src/insns/cjalr_jalr_32bit.adoc]: I reworded it. Hopefully clearer.

File: src/insns/cjalr_jalr_32bit.adoc [hunk 2]
```diff
@@ -24,11 +24,20 @@ Capability Mode Description::
...
-to `cd`.
+performed. The check for length violation is done next using the converted
+address and the original bounds.
+
+[NOTE]
+====
+When performing invalid address conversion the tag never needs to be cleared because
+the earlier length violation check is a stricter condition than the representability
+check, and the capability is always unsealed before invalid address conversion.
+====
```
Comment (andresag01) [inline src/insns/cjalr_jalr_32bit.adoc:37]: I don't fully agree with this note for a couple of reasons:
1. The tag is never cleared because the instruction excepts in all cases instead of changing PCC when something goes wrong
2. The capability is not _always_ unsealed. As @arichardson pointed out, there is an exception when the immediate is not 0 because you may be jumping to an instruction that is not the entry point.

I would prefer to remove this note altogether.

File: src/insns/cjalr_jalr_32bit.adoc [hunk 3]
```diff
@@ -24,11 +24,20 @@ Capability Mode Description::
...
+
+The capability is then installed in <<pcc>>. The <<pcc>> of the next instruction
+following the jump (<<pcc>> + 4) is sealed and written to `cd`.
```
Comment (andresag01) [inline src/insns/cjalr_jalr_32bit.adoc:40]: Please note that that this test is wrapped under the `Synopsis::` header, but it is not currently indented properly because it is after the note and it is also missing the + at the end of the previous paragraph.

File: src/insns/cjalr_jalr_32bit.adoc [hunk 4]
```diff
@@ -24,11 +24,20 @@ Capability Mode Description::
...
 target capability is obtained by unsealing `cs1` and incrementing its address by the
 sign-extended 12-bit immediate, and then setting the least-significant bit of the
-result to zero. The target capability may have
+result to zero. The target address may then have
 xref:section_invalid_addr_conv[xrefstyle=short]
-performed and is then installed in <<pcc>>. The <<pcc>>
-of the next instruction following the jump (<<pcc>> + 4) is sealed and written
-to `cd`.
+performed. The check for length violation is done next using the converted
+address and the original bounds.
```
Comment (andresag01) [inline src/insns/cjalr_jalr_32bit.adoc:30]: I think this text is better placed as a note in the `Exceptions::` section where the actual failure cases are highlighted.

**Review by andresag01 — CHANGES_REQUESTED** [review]

File: src/insns/cjalr_jalr_32bit.adoc (hunk 2 above)
Comment (Timmmm) [reply src/insns/cjalr_jalr_32bit.adoc:37]: I'll just remove this so we can get the important bit landed.

**Review by andresag01 — APPROVED** [review]
